src/Icons/src/Registry/CacheIconRegistry.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Icons\Registry;
+
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\UX\Icons\Exception\IconNotFoundException;
+use Symfony\UX\Icons\IconRegistryInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @internal
+ */
+final class CacheIconRegistry implements IconRegistryInterface, CacheWarmerInterface
+{
+    /**
+     * @param IconRegistryInterface[] $registries
+     */
+    public function __construct(private \Traversable $registries, private CacheInterface $cache)
+    {
+    }
+
+    public function get(string $name, bool $refresh = false): array
+    {
+        return $this->cache->get(
+            sprintf('ux-icon-%s', str_replace([':', '/'], ['-', '-'], $name)),
+            function () use ($name) {
+                foreach ($this->registries as $registry) {
+                    try {
+                        return $registry->get($name);
+                    } catch (IconNotFoundException) {
```

Comment:
In my mind
- we have only on registry there, and thus we should "just" decorate 
- we have multiple registries there, and then i think we should have something like "has" or "support", allowing them to be registered for specific paths / storage / etc